cannot find client.manifest.mjs when adding or removing package while dev server is up

### Environment

------------------------------
- Operating System: `Windows_NT`
- Node Version:     `v16.17.0`
- Nuxt Version:     `3.0.0-rc.11`
- Nitro Version:    `0.5.4`
- Package Manager:  `yarn@3.2.3`
- Builder:          `vite`
- User Config:      `-`
- Runtime Modules:  `-`
- Build Modules:    `-`
------------------------------

### Reproduction

- start nuxt dev 
- on another terminal , install any package using yarn (not tested with npm or pnpm)
- try to go on any page 
- error 500 

### Describe the bug

Hi :wave: ,

adding a package while the dev server is on cause `.nuxt/dev` and `.nuxt/dist` being deleted.

stopping and reloading the server solves the issue since it regenerate `.nuxt` directory.

I reproduced this on window and WSL(ubuntu)

### Additional context

_No response_

### Logs

```shell
Cannot find module '/mnt/d/GIT/test/.nuxt/dist/server/client.manifest.mjs' imported from /mnt/d/GIT/test/.nuxt/dev/index.mjs
```

--- a/packages/nuxi/src/commands/dev.ts
+++ b/packages/nuxi/src/commands/dev.ts
@@ -158,6 +158,11 @@ export default defineNuxtCommand({
 
     await load(false)
 
+    const distWatcher = chokidar.watch(resolve(rootDir, '.nuxt/dist'), { ignoreInitial: true })
+    distWatcher.on('unlinkDir', () => {
+      dLoad(true, '.nuxt/dist directory has been removed')
+    })
+
     return 'wait' as const
   }
 })

--- a/packages/nuxi/src/commands/dev.ts
+++ b/packages/nuxi/src/commands/dev.ts
@@ -59,6 +59,8 @@ export default defineNuxtCommand({
     const { loadNuxt, buildNuxt } = await loadKit(rootDir)
 
     let currentNuxt: Nuxt
+    let distWatcher: chokidar.FSWatcher
+
     const showURL = () => {
       listener.showURL({
         // TODO: Normalize URL with trailing slash within schema
@@ -75,10 +77,19 @@ export default defineNuxtCommand({
         if (currentNuxt) {
           await currentNuxt.close()
         }
+        if (distWatcher) {
+          distWatcher.close()
+        }
+
         currentNuxt = await loadNuxt({ rootDir, dev: true, ready: false })
         if (!isRestart) {
           showURL()
         }
+
+        distWatcher = chokidar.watch(resolve(currentNuxt.options.buildDir, 'dist'), { ignoreInitial: true })
+        distWatcher.on('unlinkDir', () => {
+          dLoad(true, '.nuxt/dist directory has been removed')
+        })
 
         // Write manifest and also check if we need cache invalidation
         if (!isRestart) {
@@ -157,11 +168,6 @@ export default defineNuxtCommand({
     })
 
     await load(false)
-
-    const distWatcher = chokidar.watch(resolve(rootDir, '.nuxt/dist'), { ignoreInitial: true })
-    distWatcher.on('unlinkDir', () => {
-      dLoad(true, '.nuxt/dist directory has been removed')
-    })
 
     return 'wait' as const
   }

--- a/packages/nuxi/src/commands/dev.ts
+++ b/packages/nuxi/src/commands/dev.ts
@@ -86,7 +86,7 @@ export default defineNuxtCommand({
           showURL()
         }
 
-        distWatcher = chokidar.watch(resolve(currentNuxt.options.buildDir, 'dist'), { ignoreInitial: true })
+        distWatcher = chokidar.watch(resolve(currentNuxt.options.buildDir, 'dist'), { ignoreInitial: true, depth: 0 })
         distWatcher.on('unlinkDir', () => {
           dLoad(true, '.nuxt/dist directory has been removed')
         })

--- a/packages/nuxi/src/commands/dev.ts
+++ b/packages/nuxi/src/commands/dev.ts
@@ -78,7 +78,7 @@ export default defineNuxtCommand({
           await currentNuxt.close()
         }
         if (distWatcher) {
-          distWatcher.close()
+          await distWatcher.close()
         }
 
         currentNuxt = await loadNuxt({ rootDir, dev: true, ready: false })

